src/main/java/com/direwolf20/buildinggadgets/common/util/exceptions/PaletteOverflowException.java

```diff
@@ -0,0 +1,13 @@
+package com.direwolf20.buildinggadgets.common.util.exceptions;
+
+import com.direwolf20.buildinggadgets.api.building.Region;
+
+public class PaletteOverflowException extends Exception {
+
+    private static final long serialVersionUID = 6588933909692330592L;
+
+    public PaletteOverflowException(Region region, int finalPalettes) {
```

Comment:
Effective Java#Item 63: "Include failure-capture information in detail mesages". Could you add the finalPalettes as a final field to the `PaletteOverflowException`? (with a getter of course)